subworkflows/local/db_check.nf

```diff
@@ -88,6 +89,7 @@ def validate_db_rows(LinkedHashMap row) {
 
     // check if any form of bracken params, that it must have `;`
     if ( row.tool == 'bracken' && row.db_params && !row.db_params.contains(";") )  error("[nf-core/taxprofiler] ERROR: Invalid database db_params entry. Bracken requires a semi-colon if passing parameter. Error in: ${row}")
+    if ( row.tool == 'kmcp' && row.db_params && !row.db_params.contains(";"))  error ("[nf-core/taxprofiler] ERROR: Invalid database db_params entry. KMCP requires a semi-colon if passing argument. Error in: ${row}")
```

Comment:
```suggestion
    if ( row.tool == 'kmcp' && row.db_params && row.db_params.matches(".*;\$"))  error ("[nf-core/taxprofiler] ERROR: Invalid database db_params entry. KMCP  only requires a semi-colon if passing arguments to KMCP profile, in cases of which should go after the semi-colon. Error in: ${row}")
```

Reply:
The only case we want to error is when someone tries to pass no parameters to KCMP profile, otherwise we always assume the second position arguments is for KMCP profile. If no KMCP profile arguments are needed, then we don't need the semi colon.